File: api/src/object/status/checks/env.ts
import { Checker, CheckerInput, CheckerOutput } from "../registery";

export default class implements Checker {
    VAILD_ENVIRONMENTS = [
        'production',
        'staging',
        'prod',
        'stage',
    ];

    async check(input: CheckerInput, result: CheckerOutput): Promise<void> {
        if (!this.VAILD_ENVIRONMENTS.includes(input.cacheInfo.environment)) {
            result.warning(`Environment is not set to production or staging. It is set to ${input.cacheInfo.environment}`);
            return
        }
    }
}

File: api/internal/shopware/checker/env.go
package checker

import (
	"context"
	"strings"
)

var validEnvironments = map[string]bool{
	"production": true,
	"staging":    true,
	"prod":       true,
	"stage":      true,
}

func checkEnv(_ context.Context, input Input, output *Output) {
	// An unfetched cache info would look like an empty (and therefore invalid)
	// environment name, which is a fabricated warning rather than a finding.
	if input.Missing.CacheInfo {
		output.MarkUnavailable(SourceShopware, prefixEnv)
		return
	}

	env := strings.ToLower(input.CacheInfo.Environment)
	params := map[string]any{"environment": input.CacheInfo.Environment}
	if !validEnvironments[env] {
		output.Warning(prefixEnv, "check.env.invalid", params, SourceShopware, "")
	} else {
		output.Success(prefixEnv, "check.env.valid", params, SourceShopware, "")
	}
}
